src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -460,13 +463,35 @@ static CHIP_ERROR GetAddrInfo(void * context, DnssdResolveCallback callback, uin
     protocol = kDNSServiceProtocol_IPv6;
 #endif
 
-    err = DNSServiceGetAddrInfo(&sdRef, 0 /* flags */, interfaceId, protocol, hostname, OnGetAddrInfo, sdCtx);
-    VerifyOrReturnError(CheckForSuccess(sdCtx, __func__, err, true), CHIP_ERROR_INTERNAL);
+    if (interfaceId != static_cast<uint32_t>(-1))
```

Comment:
Why not just use the defined kDNSServiceInterfaceIndexLocalOnly?

Reply:
Confirmed this compiles:

```
diff --git a/src/platform/Darwin/DnssdImpl.cpp b/src/platform/Darwin/DnssdImpl.cpp
index ff0f96ffd..f18220cbc 100644
--- a/src/platform/Darwin/DnssdImpl.cpp
+++ b/src/platform/Darwin/DnssdImpl.cpp
@@ -463,7 +463,7 @@ static CHIP_ERROR GetAddrInfo(void * context, DnssdResolveCallback callback, uin
     protocol = kDNSServiceProtocol_IPv6;
 #endif
 
-    if (interfaceId != static_cast<uint32_t>(-1))
+    if (interfaceId != kDNSServiceInterfaceIndexLocalOnly)
     {
         // -1 is the local only interface. If we're not on that interface, we need to get the address for the given hostname.
         err = DNSServiceGetAddrInfo(&sdRef, 0 /* flags */, interfaceId, protocol, hostname, OnGetAddrInfo, sdCtx);

```

Reply:
If you know what header that's defined in, we can make that change.

Reply:
It's not in our code.  It's in dns_sd.h, which we include from `platform/Darwin/DnssdImpl.h`.

https://opensource.apple.com/source/mDNSResponder/mDNSResponder-544/mDNSShared/dns_sd.h.auto.html